db/seeds.rb

```diff
@@ -67,99 +91,91 @@ def fetch_db_planets(url)
+puts "This might take a while..."
+
+1.upto(70) do |n|
+  # user #17 does not exist. Surely we might find a better solution...
+  if n == 17
+    next
+  else
+    person = serialized(Swapi.get_person(n))
+    planet = serialized(open(person["homeworld"]).read)["name"] unless person["homeworld"].empty?
+    p_id = Planet.find_by(name: planet).id unless person["homeworld"].empty?
+    specie = serialized(open(person["species"].first).read)["name"] unless person["species"].empty?
+    s_id = Specie.find_by(name: specie).id unless person["species"].empty?
+
+    User.create!(
+      name: person["name"],
+      planet_id: p_id,
+      specie_id: s_id,
+      email: Faker::Internet.email,
+      password: '123456'
+    )
+  end
+end
+
+# [SHIPS] Next steps -> generate real addresses with reverse/geocoding
+# -> fetch real prices of ships to determine rental value
+puts "[SHIPS in #{(Time.now - time).round} sec] Designing the best vehicules for our dear clients!"
 
-def fetch_db_shipsinfos(url)
-  url.each do |ships_info|
-    ShipsInfo.create(name: ships_info["model"], ship_class: ships_info["starship_class"])
+def fetch_db_ships_classes(url)
+  url.each do |ships_class|
+    ShipsClass.create(name: ships_class["starship_class"])
   end
 end
 
+def fetch_db_ships_models(url)
+  url.each do |ships_model|
+    ships_class = ShipsClass.find_by(name: ships_model["starship_class"])
+    if ships_class.nil?
+      next
+    else
+      ships_class_id = ships_class.id
+      ShipsModel.create(name: ships_model["name"], ships_class_id: ships_class_id)
+    end
+  end
+end
+
+# Why don't we use the gem -> gem 'tatooine' or others?
 starships = serialized(open("http://swapi.co/api/starships/").read)
 while starships["next"] != nil
-  fetch_db_shipsinfos(starships["results"])
+  fetch_db_ships_classes(starships["results"])
+  fetch_db_ships_models(starships["results"])
   starships = serialized(open(starships["next"]).read)
 end
 
 lastship = serialized(open("http://swapi.co/api/starships/?page=4").read)
-fetch_db_shipsinfos(lastship["results"])
-
-#Create 10 users and get goods planets and species from API and ids from table
-
-1.upto(10) do |n|
-  person = serialized(Swapi.get_person(n))
-  planet = serialized(open(person["homeworld"]).read)["name"]
-  p_id = Planet.find_by(name: planet).id
-  specie = serialized(open(person["species"].first).read)["name"]
-  s_id = Specie.find_by(name: specie).id
-
-  User.create!(
-    name: person["name"],
-    planet_id: p_id,
-    specie_id: s_id,
-    email: Faker::Internet.email,
-    password: '123456'
+fetch_db_ships_classes(lastship["results"])
+fetch_db_ships_models(lastship["results"])
+
+1.upto(50) do |n|
+  Ship.create(
+    name: Faker::Space.nasa_space_craft,
+    address: Faker::Address.country,
+    price: rand(10000..100000),
+    ships_model_id: random(ShipsModel),
+    user_id: random(User)
+  )
+end
+
+# [BOOKINGS] -> I modified the generator to be sure that the ship is associated with its owner
```

Comment:
Actually... I did not and deleted this change. Was confused between owner/renter...